src/core/utils/getAddressesBalances.ts

```diff
@@ -34,11 +41,27 @@ export async function getAddressesBalances({
     throw new Error('Multicall balance checker not supported for this chain')
   }
 
-  const contract = new Contract(contractAddress, BalanceCheckerABI, provider)
+  const contract = BalanceChecker__factory.connect(contractAddress, provider)
 
   const balances = await contract.balances(addresses, tokens, {
     blockTag: blockNumber,
   })
 
-  return formatAddressBalances<BigNumber>(balances, addresses, tokens)
+  return formatAddressBalances(balances, addresses, tokens)
+}
+
+function formatAddressBalances(
+  values: BigNumber[],
+  addresses: string[],
+  tokens: string[],
+): AddressBalanceMap {
+  const balances: AddressBalanceMap = {}
+  addresses.forEach((addr, addrIdx) => {
+    balances[addr] = {}
+    tokens.forEach((tokenAddr, tokenIdx) => {
+      const balance = values[addrIdx * tokens.length + tokenIdx]
+      balances[addr]![tokenAddr] = balance!.toString()
```

Comment:
can the balance request ever fail?

What happens when one fails? does the whole thing fail?

Reply:
Yes. If getting the balance for any token fails, the whole contract call fails (https://etherscan.io/address/0xb1f8e55c7f64d203c1400b9d8555d050f94adf39#code).

I copied this function from their library and just added the typescript bits.